roles/change-paths/tasks/main.yml

```diff
@@ -0,0 +1,167 @@
+  set_fact:
+    absolute_vhost_path_sites_service: "{{ vhost }}.stanford.edu"
+
+- name: Set absolute non-vhost path from sites service
+  set_fact:
+    absolute_path_sites_service: "{{ sites_service }}.stanford.edu/{{ inventory_hostname }}"
+
+- name: Set absolute path for testing
+  set_fact:
+    absolute_path_testing: "{% if vhost is defined %}{{ vhost }}{% else %}{{ acsf_site_name }}{% endif %}-{{ acsf_environment }}.cardinalsites.stanford.edu" 
+
+- name: Set absolute path for production
+  set_fact:
+    absolute_path_prod: "{% if vhost is defined %}{{ vhost }}{% else %}{{ acsf_site_name }}{% endif %}.stanford.edu"
+
+- name: Set new absolute path based on destination
+  set_fact:
+    new_absolute_path: "{% if acsf_environment == '' %}{{ absolute_path_prod }}{% else %}{{ absolute_path_testing }}{% endif %}"
+
+- name: Debug absolute path for site
+  debug:
+    msg: "Absolute path: {{ new_absolute_path }}"
+
+- name: Set drush sar command
+  set_fact:
+    drush_sar_command: "drush @acsf{{ drush_environment }}.{{ stack }}.{{ acsf_site_name }}" 
+
+# See discussion in https://github.com/SU-SWS/ansible-playbooks/pull/17/files
+# And https://github.com/SU-SWS/ansible-playbooks/pull/22/files
+- name: Update full paths in text
+  shell: "{{ drush_sar_command }} sar -y '{{ item }}{{ absolute_path_sites_service }}/' '{{ item }}{{ new_absolute_path }}/'"
+  with_items:
+   - "{{ protocols }}"
+
+# We may want to exclude sar from this
+- name: Remove absolute paths for links and menus
+  shell: "{{ drush_sar_command }} {{ item[0] }} -y '{{ item[1] }}{{ absolute_path_sites_service }}/' ''"
+  with_nested:
+    - [ "sarl", "sarm" ]
+    - "{{ protocols }}"
+  notify: Clear site cache
+
+# These are separated out because we will only be using sar, not all sar_commands
+- name: Remove absolute paths for link attributes
+  shell: "{{ drush_sar_command }} sar -y '{{ item[0] }}/{{ absolute_path_sites_service }}/' ''"
+  with_together:
+    - "{{ link_attributes }}"
+  notify: Clear site cache
+
+# Have to run these again for links that might include vhosts
+- name: Update full vhost paths in text
+  shell: "{{ drush_sar_command }} sar -y '{{ item }}{{ absolute_vhost_path_sites_service }}/' '{{ item }}{{ new_absolute_path }}/'"
+  with_items:
+   - "{{ protocols }}"
+  when: vhost is defined
+
+# We may want to exclude sar from this
+- name: Remove absolute vhost paths for links and menus
+  shell: "{{ drush_sar_command }} {{ item[0] }} -y '{{ item[1] }}{{ absolute_vhost_path_sites_service }}/' ''"
+  with_nested:
+    - [ "sarl", "sarm" ]
+    - "{{ protocols }}"
+  notify: Clear site cache
+  when: vhost is defined
+
+# These are separated out because we will only be using sar, not all sar_commands
+- name: Remove absolute vhost paths for link attributes
+  shell: "{{ drush_sar_command }} sar -y '{{ item[0] }}/{{ absolute_vhost_path_sites_service }}/' '{{ item[0] }}/'"
+  with_together:
+    - "{{ link_attributes }}"
```

Comment:
Also, this needs to be fixed. See https://github.com/SU-SWS/ansible-playbooks/blob/master/roles/upload-site/tasks/main.yml#L129.

We want to replace `href="/jbickar/"` with `href="/"`

Reply:
Oh, you do that below. I *think* that this task is not needed, because you're looking for `href="methane.stanford.edu/...`, and that shouldn't be in an HTML text field.

Reply:
I removed 112, but kept 89.  Is that we decided was the right course of action?

Reply:
I added it back; see [this TODO](https://github.com/SU-SWS/ansible-playbooks/pull/26/files#diff-22f9b66efba0529e7d1194e46abd26ddR143)